### ISSUE TITLE
Feature Request: Paginate User Management page

### ISSUE THREAD

**Issue by hexxamillion** [nilsteampassnet/TeamPass#810]
Can the User Management page be paginated? This would be helpful when there are a lot of users and prevent the page from taking a long time to load.

**Comment by hexxamillion** [comment]
I currently have 100+ users so pagination would be awesome. Additionally I see there is now a search field on 2.1.22 so that's cool. With pagination and a search field I think that will be the ultimate.

**Comment by hexxamillion** [comment]
I tried the indexes as suggested. That did not improve anything. Ideally from a web dev perspective you want pagination as this list of users will only grow. No one wants to scroll 100's of users.

**Comment by hexxamillion** [comment]
I was looking into tinkering around with this as it looks like the datatables library is used by teampass. That library has pagination. I cannot find where the datatable library is referenced. I would think this could be modified by just adding one of the pagination plugin functions for datatables.

**Comment by hexxamillion** [comment]
This seems to really only need to be applied for the Admin user since they see all users. The Manager role only has a subset and probably has no more than a few users they manage.

**Comment by hexxamillion** [comment]
I think I have found out a the solution for this. This works great for admin but for managers the user list logic may need to be looked at. It doesn't seem to display all on the same page. This only happens under user manager. Admin is perfect. It would be great if this feature can be added to the next version.

I edited the users.php. The result is on the user management page it only displays 10 at a time and there are NEXT and PREVIOUS links on the top and bottom of the user list table.

I would like to propose adding this before the user list query on line ~94 or 95
```
//LIMIT Results code
//check if the starting row variable was passed in the URL or not
if (!isset($_GET['startrow']) or !is_numeric($_GET['startrow'])) {
//we give the value of the starting row to 0 because nothing was found in URL
   $startrow = 0;
//otherwise we take the value from the URL
} else {
   $startrow = (int)$_GET['startrow'];
 }
 $prev = $startrow - 10;
//END LIMIT results code 
```

Then in the query code for user list I commented out original and added:
```
// Get through all users
//$rows = DB::query("SELECT * FROM ".$pre."users ORDER BY login ASC");
$rows = DB::query("SELECT * FROM ".$pre."users ORDER BY login ASC LIMIT $startrow, 10");
```

Then way at the bottom of the table I added this code on line ~326
```
//only print a "Previous" link if a "Next" was clicked
if ($prev >= 0)
     echo '<a href="'.$_SERVER['PHP_SELF'].'?page=manage_users&startrow='.$prev.'">Previous</a> |';
     echo ' <a href="'.$_SERVER['PHP_SELF'].'?page=manage_users&startrow='.($startrow+10).'">Next</a>';

echo '
            </tbody>
        </table>';
if ($prev >= 0)
     echo '<a href="'.$_SERVER['PHP_SELF'].'?page=manage_users&startrow='.$prev.'">Previous</a> |';
     echo ' <a href="'.$_SERVER['PHP_SELF'].'?page=manage_users&startrow='.($startrow+10).'">Next</a>';
```
![next](https://cloud.githubusercontent.com/assets/1843715/7756457/342c53e2-ffb0-11e4-9725-5ae5a1938789.jpg)
![prevnext](https://cloud.githubusercontent.com/assets/1843715/7756460/3b2cb57e-ffb0-11e4-9b3d-e8c63833b2a3.jpg)

**Comment by hexxamillion** [comment]
This also seems to impact the search functionality. If the user does not exist on the current results then it will not find it. The search should not be limited to just result on the page but in the user table.

**Comment by hexxamillion** [comment]
This can be closed as the feature is available as on 2.1.24. Thanks!
